clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -502,7 +503,16 @@ public synchronized void dropStorePartition(
     Set<Integer> remainingPartitions = storageEngine.getPartitionIds();
     LOGGER.info("Dropped partition {} of {}, remaining partitions={}", partition, kafkaTopic, remainingPartitions);
 
+    // Check if blob transfer is enabled and if there are ongoing transfers
```

Comment:
IIUC this can still have issue:
p1(blob transfer thread): (1)SE.markBlob()
p2(drop partition thread): (1)SE.hasBlob() -> (2) SE.markBlob()
Now we are competing the object lock of SE. but if we do p2.(1)->p1.(1)->p2.(2), wouldn't this fall into the same case?

Can we make hasBlob and markBlob in the same action? For example, we can return a True/False control whether we actually mark it for drop and whether we actually can drop it.